pandas/core/generic.py

```diff
@@ -7709,9 +7709,11 @@ def resample(
         convention: str = "start",
         kind: Optional[str] = None,
         loffset=None,
-        base: int = 0,
+        base: Optional[int] = None,
         on=None,
         level=None,
+        origin=None,
```

Comment:
```suggestion
        origin: pd.Timestamp = None,
```

Ideally should annotate all new arguments

Reply:
Yeah this is a *very* good idea ! however I think we need to decide the proper type before doing that. More on this here: https://github.com/pandas-dev/pandas/pull/31809#discussion_r406554007